src/Schema/Events/Tap.ts

```diff
@@ -235,3 +236,27 @@ export interface TappedPromoSpace {
   destination_path: string
   subject: string
 }
+
+/**
+ * A user taps an icon on the tab bar
+ *
+ * This schema describes events sent to Segment from [[tappedTabBar]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "tappedTabBar",
+ *    badge: true
+ *    context_module: "tabBar",
+ *    context_screen_owner_type: "home",
+ *    tab: "messages"
+ *  }
+ * ```
+ */
+export interface TappedTabBar {
+  action: ActionType.tappedTabBar
+  badge?: boolean
```

Comment:
up for debate whether we should have this, so I've left it optional, but my thinking is: "What if we want to know a user is tapping an icon with a badge on it?" Do we foresee this being important? Otherwise, I'm ok to remove this property throughout.

Reply:
Let's make this required, we can set the default to "false" in the helper so that this is always included.